src/main/java/frc/robot/commands/DriveStraight.java

```diff
@@ -0,0 +1,180 @@
+   * @param translation A {@link Translation2d} instance describing the line on which to travel.
+   *     This is a vector relative to the current position.
+   * @param maxSpeed The maximum speed at which to travel.
+   */
+  public DriveStraight(Swerve drivetrain, Translation2d translation, double maxSpeed) {
+    this(drivetrain, () -> translation, maxSpeed, () -> drivetrain.getPosition().getRotation());
+  }
+
+  /**
+   * Creates a new DriveStraight that drives robot along the specified vector while rotating the
+   * robot to the desired orientation.
+   *
+   * @param drivetrain The {@link Swerve} representing the robot drivetrain.
+   * @param translation A {@link Translation2d} instance describing the line on which to travel.
+   *     This is a vector relative to the current position.
+   * @param maxSpeed The maximum speed at which to travel.
+   * @param orientation The desired orientation at the end of the command.
+   */
+  public DriveStraight(
+      Swerve drivetrain, Translation2d translation, double maxSpeed, Rotation2d orientation) {
+    this(drivetrain, () -> translation, maxSpeed, () -> orientation);
+  }
+
+  /**
+   * Creates a new DriveStraight that drives robot to the specified absolute location and
+   * orientation on the field.
+   *
+   * @param drivetrain The {@link Swerve} representing the robot drivetrain.
+   * @param position A {@link Pose2d} instance describing the absolute position and orientation to
+   *     drive to.
+   * @param maxSpeed The maximum speed at which to travel.
+   */
+  public DriveStraight(Swerve drivetrain, Pose2d position, double maxSpeed) {
+    this(
+        drivetrain,
+        () -> position.getTranslation().minus(drivetrain.getPosition().getTranslation()),
+        maxSpeed,
+        () -> position.getRotation());
+  }
+
+  /**
+   * Constructs an instance of this class.
+   *
+   * @param drivetrain The {@link Swerve} representing the robot drivetrain.
+   * @param translationSupplier Supplies a {@link Translation2d} instance describing the line on
+   *     which to travel. This is a vector relative to the current position.
+   * @param maxSpeed The maximum speed at which to travel.
+   * @param orientationSupplier Supplies the desired orientation at the end of the command.
+   */
+  private DriveStraight(
+      Swerve drivetrain,
+      Supplier<Translation2d> translationSupplier,
+      double maxSpeed,
+      Supplier<Rotation2d> orientationSupplier) {
+    this.drivetrain = drivetrain;
+    this.translationSupplier = translationSupplier;
+    this.controller = drivetrain.createDriveController();
+    this.maxSpeed = maxSpeed;
+    this.orientationSupplier = orientationSupplier;
+
+    addRequirements(drivetrain);
+  }
+
+  @Override
+  public void initialize() {
+    initialPose = drivetrain.getPosition();
+
+    Translation2d translation = translationSupplier.get();
+    distance = translation.getNorm();
```

Comment:
distance can be a local var

Reply:
This is defined as a local variable on line 140 but should be defined here for clarity.

```suggestion
    double distance = translation.getNorm();
```